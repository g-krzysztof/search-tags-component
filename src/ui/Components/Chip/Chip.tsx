import React from 'react'
import { Box } from '../Box'
import { IconButton } from '../IconButton'
import { Text } from '../Text'

interface ChipProps {
  label: string
  onClick: () => void
}

export const Chip: React.FC<ChipProps> = ({ label, onClick }) => {
  return (
    <Box
      border="grayLightThin"
      pl="xxs"
      pr="xxxs"
      py="xxxs"
      borderRadius="rounded"
      display="flex"
      justifyContent="space-between"
      alignItems="center"
    >
      <Text color="black" fontSize="xm" fontWeight="bold">
        {label}
      </Text>
      <Box pl="xxxs">
        <IconButton
          iconName="close"
          variant="bare"
          onClick={onClick}
          color="grayMedium"
        />
      </Box>
    </Box>
  )
}
