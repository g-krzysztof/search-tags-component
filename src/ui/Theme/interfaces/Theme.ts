import {
  Borders,
  Colors,
  FontSizes,
  FontWeights,
  LineHeights,
  Motion,
  Radii,
  Space,
} from '.'

export interface Theme {
  borders: Borders
  colors: Colors
  fontSizes: FontSizes
  fontWeights: FontWeights
  lineHeights: LineHeights
  motion: Motion
  radii: Radii
  space: Space
}
