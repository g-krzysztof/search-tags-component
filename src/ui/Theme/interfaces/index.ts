export type { Borders } from './Borders'
export type { Colors } from './Colors'
export type { FontSizes } from './FontSizes'
export type { FontWeights } from './FontWeights'
export type { LineHeights } from './LineHeights'
export type { Motion } from './Motion'
export type { Radii } from './Radii'
export type { Space } from './Space'
export type { Theme } from './Theme'
