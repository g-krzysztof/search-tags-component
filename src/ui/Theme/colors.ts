import { Colors } from './interfaces'

export const colors: Colors = {
  blue: '#2463ec',
  redDark: '#c22f3f',
  redLight: '#ff686f',
  greenDark: '#3b8b58',
  greenLight: '#1ea24f',
  purple: '#a461e5',
  grayLight: '#f4f4f6',
  grayMedium: '#a5a4a9',
  grayDark: '#626264',
  black: '#161616',
  white: '#ffffff',
}
