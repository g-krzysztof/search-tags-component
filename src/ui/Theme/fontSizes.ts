import { FontSizes } from './interfaces'

// values should be interpreted as rem
// rem base: 1rem = 16px

export const fontSizes: FontSizes = {
  xs: 10,
  s: 12,
  m: 14,
  l: 16,
  xl: 18,
  xxl: 32,
}
