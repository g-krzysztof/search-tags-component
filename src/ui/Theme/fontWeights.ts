import { FontWeights } from './interfaces'

export const fontWeights: FontWeights = {
  regular: 400,
  medium: 500,
  bold: 700,
}
