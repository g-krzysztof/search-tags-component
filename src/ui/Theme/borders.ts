import { Borders } from './interfaces'
import { colors } from './colors'

export const borders: Borders = {
  // solid
  grayLight: `2px solid ${colors.grayLight}`,
  grayLightThin: `1px solid ${colors.grayLight}`,
  grayMedium: `2px solid ${colors.grayMedium}`,
  grayMediumThin: `1px solid ${colors.grayMedium}`,
  blueMedium: `2px solid ${colors.blue}`,
  whiteMedium: `2px solid ${colors.white}`,
}
