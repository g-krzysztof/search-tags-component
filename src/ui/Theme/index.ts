import { borders } from './borders'
import { colors } from './colors'
import { fontSizes } from './fontSizes'
import { fontWeights } from './fontWeights'
import { Theme } from './interfaces'
import { lineHeights } from './lineHeights'
import { motion } from './motion'
import { radii } from './radii'
import { space } from './space'

export { borders } from './borders'
export { colors } from './colors'
export { fontSizes } from './fontSizes'
export { fontWeights } from './fontWeights'
export type { Theme } from './interfaces'
export { lineHeights } from './lineHeights'
export { motion } from './motion'
export { radii } from './radii'
export { space } from './space'

export const theme: Theme = {
  borders,
  colors,
  fontSizes,
  fontWeights,
  lineHeights,
  motion,
  radii,
  space,
}
