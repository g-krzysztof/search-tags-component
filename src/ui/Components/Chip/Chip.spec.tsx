import { render, fireEvent, getByTestId } from '@testing-library/react'
import '@testing-library/jest-dom'
import { ThemeProvider } from 'styled-components'
import { theme } from '../../Theme'
import { Chip } from './Chip'

describe('Chip', () => {
  const mockOnClick = jest.fn()

  it('renders the chip with the provided label', () => {
    const { getByText } = render(
      <ThemeProvider theme={theme}>
        <Chip label="Test Chip" onClick={mockOnClick} />
      </ThemeProvider>,
    )
    expect(getByText('Test Chip')).toBeInTheDocument()
  })

  it('calls the onClick function when the close icon is clicked', () => {
    const { getByTestId } = render(
      <ThemeProvider theme={theme}>
        <Chip label="Test Chip" onClick={mockOnClick} />
      </ThemeProvider>,
    )
    fireEvent.click(getByTestId('iconButton'))
    expect(mockOnClick).toHaveBeenCalled()
  })
  it('renders Chip unchanged', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Chip label="Test Chip" onClick={mockOnClick} />
      </ThemeProvider>,
    )
    expect(container).toMatchSnapshot()
  })
})
