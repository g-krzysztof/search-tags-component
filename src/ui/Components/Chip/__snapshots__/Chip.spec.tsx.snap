// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip renders Chip unchanged 1`] = `
<div>
  <div
    alignitems="center"
    border="grayLightThin"
    borderradius="rounded"
    class="sc-8b3de090-0 hpPain"
    display="flex"
    justifycontent="space-between"
    pl="xxs"
    pr="xxxs"
    py="xxxs"
  >
    <div
      class="sc-5a158f7a-0 hHFrbZ"
      color="black"
      font-size="xm"
      font-weight="bold"
    >
      Test Chip
    </div>
    <div
      class="sc-8b3de090-0 cFhTLI"
      pl="xxxs"
    >
      <div
        alignitems="center"
        borderradius="rounded"
        class="sc-8b3de090-0 kHVuUD sc-628c0d2-0 jCjijW"
        data-testid="iconButton"
        display="flex"
        height="20px"
        justifycontent="center"
        variant="bare"
        width="20px"
      >
        <div
          alignitems="center"
          class="sc-8b3de090-0 NeWgB sc-b381ab80-0 hzQRJe"
          color="grayMedium"
          data-testid="close"
          display="flex"
          justifycontent="center"
          size="24"
        >
          <svg
            role="img"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
